### PR TITLE
chore(build): Add bundlesize check script

### LINKED ISSUE #2: Use bundlesize for dist bundle
When the project is further along and we have a dist bundle we should leverage [`bundlesize`](https://github.com/siddharthkp/bundlesize) to keep the bundle size in check.

### PR DESCRIPTION
**Pull request by ryanwholey**
This tool should be really helpful for us as we continue to grow! I was actually surprised that our bundle is already over 3k (3.14 to be exact π). We'll need to add our travis token to if we want it to show up as a badge on github as outlined in the [bundlesize readme](https://github.com/siddharthkp/bundlesize)

fixes #2

### CODE REVIEW

**Comment by benmvp** [comment]
So I don't see bundlesize as a test, but more of a validation if that makes sense. I think `validate` should definitely call `bundlesize` but I don't think `test` should. You could also add a `pretest:size` (or whatever you end up naming the script) that runs the `gulp build:dist:min` to make sure that the freshest bundle is there.

What do ya think?

Also once we set up an actual release process, we'll have the travis token and can do all the badge fun 😄

**Comment by ryanwholey** [comment]
@BenAtEventbrite ah interesting! I think i see now, I thought that we were running _only_ test:ci on travis, were actually running validate & build. I think it makes sense to move it to validate with something like pretest:size to run the build process.

File: package.json [hunk 1]
```diff
@@ -32,13 +32,16 @@
         "precommit": "lint-staged",
         "gen:declarations": "tsc --p ./tsconfig.build.json",
         "build": "npm-run-all build:targets gen:declarations",
+        "build:dist": "gulp build:dist:min",
         "build:targets": "gulp build",
         "prebuild:targets": "rm -rf dist && rm -rf lib",
         "tsc": "tsc",
         "test": "jest --config=jest.json",
+        "pretest:size": "yarn build:dist",
+        "test:size": "bundlesize",
```
[the remark this answers is not in the document]
Comment (ryanwholey) [reply package.json]: ah yep i'm with you both on that

File: package.json [hunk 2]
```diff
@@ -32,13 +32,16 @@
         "precommit": "lint-staged",
         "gen:declarations": "tsc --p ./tsconfig.build.json",
         "build": "npm-run-all build:targets gen:declarations",
+        "build:dist": "gulp build:dist:min",
         "build:targets": "gulp build",
         "prebuild:targets": "rm -rf dist && rm -rf lib",
         "tsc": "tsc",
         "test": "jest --config=jest.json",
+        "pretest:size": "yarn build:dist",
```
[the remark this answers is not in the document]
Comment (ryanwholey) [reply package.json]: sure!

**Comment by ryanwholey** [comment]
@BenAtEventbrite ah yep thatd do it. i've seen other libs move `isomorphic-fetch` to be an external dependency, we will see how it goes!

**Pull request merged**